cpp/style.rst

```diff
@@ -195,17 +195,17 @@ The capitalization rule for class names should be all words in the name capitali
 
 .. _style-guide-cpp-3-2:
 
-3-2. Variable names MUST be in mixed case starting with lower case.
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+3-2. Variable names MUST be camelCase or snake_case.
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 .. code-block:: cpp
 
    int lineWidth;
+   // OR, in primarily new code:
+   int line_width;
 
-Common practice in the C++ development community.
-Makes variables easy to distinguish from types, and effectively resolves potential naming collision as in the declaration Line line.
```

Comment:
Is it worth keeping this explanation?

Reply:
My sense of this - and the other block of reasoning I removed below - was that in this aspect our C++ style guide is now just a reflection of our Python style guide, because Python is now really our primary language and there's no good reason for the languages to differ on this aspect of style.  I'm sure the same general-best-practice arguments are ultimately behind the Python style choices, too, but as we delegate those "why" questions in Python to PEP8, it felt weird to continue enumerating them here.